GvasFormat/Serialization/BinaryReaderEx.cs

```diff
@@ -16,9 +16,15 @@ public static string ReadUEString(this BinaryReader reader)
             if (length == 0)
                 return null;
 
+            if (length < 0)
```

Comment:
this doesn't make sense; where is this atually used? as far as I can remember, ue4 serialization is forward-only, so if it ever needs to seek back, it's probably a bug or some unimplemented feature

Reply:
Yeah I had this problem with F1 Manager 2022, but you may be right ... It may be a bug

Reply:
I'll pull other changes, but this looks so wrong and this is for strings too, you'll get garbage in the beginning and will probably read into garbage at the end, so I'd ask to updata the pr without this change

Reply:
Updated it